lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -236,6 +240,46 @@ defaultAddressPoolGap =
                                  Address Pool
 -------------------------------------------------------------------------------}
 
+data ParentContext (chain :: Role) (key :: Depth -> * -> *) where
+    ParentContextUtxoExternal
+        :: key 'AccountK XPub
+        -> ParentContext 'UtxoExternal key
+
+    ParentContextUtxoInternal
+        :: key 'AccountK XPub
+        -> ParentContext 'UtxoInternal key
+
+    ParentContextMutableAccount
+        :: key 'AccountK XPub
+        -> ParentContext 'MutableAccount key
```

Comment:
I don't think there's any use-case for this one, is it?

Reply:
yeah, removed in  https://github.com/input-output-hk/cardano-wallet/pull/2508/commits/37e818cc20a5248c85f8645e712295541941e614